nagios.go

```
@@ -0,0 +1,113 @@
+	return func() error {
+		stdout, stdoutCloser, err := outFor(project, "nagios")
+		if err != nil {
+			return err
+		}
+		defer stdoutCloser.Close()
+
+		stderr, stderrCloser, err := errOutFor(project, "nagios")
+		if err != nil {
+			return err
+		}
+		defer stderrCloser.Close()
+
+		var stdoutBuff bytes.Buffer
+
+		nagiosPod, err := getProjectPodNamesBySubstr(project, "nagios")
+		if err != nil {
+			fmt.Fprint(stderr, err)
+			return err
+		}
+
+		if len(nagiosPod) == 0 {
+			err := errors.New("could not find a nagios pod in the project '" + project + "'")
+			fmt.Fprint(stderr, err)
+			return err
+		}
+
+		cmd := cmdFactory(project, nagiosPod[0])
+
+		if err := runCmdCaptureOutput(cmd, &stdoutBuff, stderr); err != nil {
+			fmt.Fprint(stderr, err)
+			return err
+		}
+
+		jsonStatus := parseNagiosStatus(string(stdoutBuff.Bytes()))
+		stdout.Write(jsonStatus)
+
+		return nil
+	}
+}
+
+// parseNagiosStatus takes the raw nagios status string and converts
+// it into a string of JSON
+func parseNagiosStatus(raw string) []byte {
+	conf := map[string][]map[string]string{}
+	currentId := ""
+	lines := strings.Split(raw, "\n")
+	attrParser := regexp.MustCompile(`[ ]*([\w]+)=([\w\d]*)`)
+	idParser := regexp.MustCompile("([\\w]+) {")
+
+	curConf := map[string]string{}
+
+	for _, line := range lines {
+		//skip comments
+		if strings.HasPrefix(line, "#") {
+			continue
+		}
+
+		ids := idParser.FindStringSubmatch(line)
+		if len(ids) > 0 {
+			currentId = ids[1]
+			if _, ok := conf[currentId]; ok == false {
+				conf[currentId] = []map[string]string{}
+			}
+			curConf = map[string]string{}
+			continue
+		}
+
+		attributes := attrParser.FindStringSubmatch(line)
+		if len(attributes) >= 3 {
+			curConf[attributes[1]] = attributes[2]
+			continue
+		}
+
+		if strings.TrimSpace(line) == "}" {
+			conf[currentId] = append(conf[currentId], curConf)
+			continue
+		}
+	}
+
+	retVal, _ := json.MarshalIndent(conf, "", " ")
```

Why ignore the error?